op-deployer/pkg/deployer/state/intent.go

```diff
@@ -32,35 +33,313 @@ func (d DeploymentStrategy) Check() error {
 	}
 }
 
+type IntentConfigType string
+
+const (
+	IntentConfigTypeTest              IntentConfigType = "test"
+	IntentConfigTypeStandard          IntentConfigType = "standard"
+	IntentConfigTypeCustom            IntentConfigType = "custom"
+	IntentConfigTypeStrict            IntentConfigType = "strict"
+	IntentConfigTypeStandardOverrides IntentConfigType = "standard-overrides"
+	IntentConfigTypeStrictOverrides   IntentConfigType = "strict-overrides"
+)
+
 var emptyAddress common.Address
+var emptyHash common.Hash
 
 type Intent struct {
-	DeploymentStrategy DeploymentStrategy `json:"deploymentStrategy" toml:"deploymentStrategy"`
+	DeploymentStrategy    DeploymentStrategy `json:"deploymentStrategy" toml:"deploymentStrategy"`
+	IntentConfigType      IntentConfigType   `json:"intentConfigType" toml:"intentConfigType"`
+	L1ChainID             uint64             `json:"l1ChainID" toml:"l1ChainID"`
+	SuperchainRoles       *SuperchainRoles   `json:"superchainRoles" toml:"superchainRoles,omitempty"`
+	FundDevAccounts       bool               `json:"fundDevAccounts" toml:"fundDevAccounts"`
+	UseInterop            bool               `json:"useInterop" toml:"useInterop"`
+	L1ContractsLocator    *artifacts.Locator `json:"l1ContractsLocator" toml:"l1ContractsLocator"`
+	L2ContractsLocator    *artifacts.Locator `json:"l2ContractsLocator" toml:"l2ContractsLocator"`
+	Chains                []*ChainIntent     `json:"chains" toml:"chains"`
+	GlobalDeployOverrides map[string]any     `json:"globalDeployOverrides" toml:"globalDeployOverrides"`
+}
+
+type SuperchainRoles struct {
+	ProxyAdminOwner       common.Address `json:"proxyAdminOwner" toml:"proxyAdminOwner"`
+	ProtocolVersionsOwner common.Address `json:"protocolVersionsOwner" toml:"protocolVersionsOwner"`
+	Guardian              common.Address `json:"guardian" toml:"guardian"`
+}
+
+var ErrSuperchainRoleZeroAddress = fmt.Errorf("SuperchainRole is set to zero address")
 
-	L1ChainID uint64 `json:"l1ChainID" toml:"l1ChainID"`
+func (s *SuperchainRoles) CheckNoZeroAddresses() error {
+	val := reflect.ValueOf(*s)
+	typ := reflect.TypeOf(*s)
 
-	SuperchainRoles *SuperchainRoles `json:"superchainRoles" toml:"superchainRoles,omitempty"`
+	// Iterate through all the fields
+	for i := 0; i < val.NumField(); i++ {
+		fieldValue := val.Field(i)
+		fieldName := typ.Field(i).Name
+
+		if fieldValue.Interface() == (common.Address{}) {
+			return fmt.Errorf("%w: %s", ErrSuperchainRoleZeroAddress, fieldName)
+		}
+	}
+	return nil
+}
+
+func (c *Intent) L1ChainIDBig() *big.Int {
+	return big.NewInt(int64(c.L1ChainID))
+}
 
-	FundDevAccounts bool `json:"fundDevAccounts" toml:"fundDevAccounts"`
+func (c *Intent) ValidateIntentConfigType() error {
```

Comment:
Why can't we put all this logic in `Check`? Why a separate validator?

Reply:
Good idea. Will move this logic into `Check`